aiida/cmdline/params/options/commands/code.py

```diff
@@ -68,11 +80,39 @@ def validate_label_uniqueness(ctx, _, value):
     return value
 
 
+ON_CONTAINER = OverridableOption(
+    '--on-container/--not-on-container',
+    is_eager=False,
+    default=False,
+    help='Whether the code is installed on the container.'
+)
+
+IMAGE = OverridableOption(
+    '--image',
+    prompt='image name',
+    required_fn=is_on_container,
+    prompt_fn=is_on_container,
+    cls=InteractiveOption,
+    help='image name.'
+)
+
+CONTAINER_CMDLINE_TMPL = OverridableOption(
+    '--container-cmdline-tmpl',
+    prompt='container cmdline params tmpl',
+    default='sarus run --mount=src=$PWD,dst=/workir,type=bind --workdir /workdir {image}',
```

Comment:
I believe this should default to singularity instead, since it is way more popular.

I would perhaps call this `container-engine-command` (I don't think "template" needs to be part of the name; we also don't call the work directory "template")